lib/src/repo.rs

```diff
@@ -1263,20 +1264,25 @@ impl MutableRepo {
         callback: impl FnMut(CommitRewriter) -> BackendResult<()>,
     ) -> BackendResult<()> {
         let options = RewriteRefsOptions::default();
-        self.transform_descendants_with_options(roots, &options, callback)
+        self.transform_descendants_with_options(roots, HashMap::new(), &options, callback)
     }
 
     /// Rewrite descendants of the given roots with options.
     ///
+    /// If a commit is in the `new_parents_map` is provided, it will be rebased
+    /// onto the new parents provided in the map instead of its original
+    /// parents.
+    ///
     /// See [`Self::transform_descendants()`] for details.
     pub fn transform_descendants_with_options(
         &mut self,
         roots: Vec<CommitId>,
+        new_parents_map: HashMap<CommitId, Vec<CommitId>>,
```

Comment:
nit: pass `new_parents_map` by reference

Reply:
Removed the function.